substrafl_examples/go_further/plot_substrafl_sklearn_fedavg.py

```diff
@@ -0,0 +1,543 @@
+from substra.sdk.schemas import AlgoSpec
+from substra.sdk.schemas import AssetKind
+
+permissions_metric = Permissions(
+    public=False,
+    authorized_ids=[ALGO_ORG_ID] + DATA_PROVIDER_ORGS_ID,
+)
+
+inputs_metrics = [
+    AlgoInputSpec(
+        identifier="datasamples",
+        kind=AssetKind.data_sample,
+        optional=False,
+        multiple=True,
+    ),
+    AlgoInputSpec(
+        identifier="opener",
+        kind=AssetKind.data_manager,
+        optional=False,
+        multiple=False,
+    ),
+    AlgoInputSpec(
+        identifier="predictions",
+        kind=AssetKind.model,
+        optional=False,
+        multiple=False,
+    ),
+]
+
+outputs_metrics = [
+    AlgoOutputSpec(
+        identifier="performance",
+        kind=AssetKind.performance,
+        multiple=False,
+    )
+]
+
+objective = AlgoSpec(
+    inputs=inputs_metrics,
+    outputs=outputs_metrics,
+    name="Accuracy",
+    description=assets_directory / "metric" / "description.md",
+    file=assets_directory / "metric" / "metrics.zip",
+    permissions=permissions_metric,
+)
+
+METRICS_DOCKERFILE_FILES = [
+    assets_directory / "metric" / "iris_metrics.py",
+    assets_directory / "metric" / "Dockerfile",
+]
+
+archive_path = objective.file
+with zipfile.ZipFile(archive_path, "w") as z:
+    for filepath in METRICS_DOCKERFILE_FILES:
+        z.write(filepath, arcname=filepath.name)
+
+metric_key = clients[ALGO_ORG_ID].add_algo(objective)
+
+# %%
+# Specify the machine learning components
+# ***************************************
+#
+# SubstraFL can be used with any machine learning framework. The framework
+# dependent functions are written in the `Algorithm <substrafl_doc/api/algorithms:Base Class>`_ object.
+#
+# In this section, you will:
+#
+# - register a model and its dependencies
+# - write your own Sklearn SubstraFL algorithm
+# - specify the federated learning strategy
+# - specify the organizations where to train and where to aggregate
+# - specify the organizations where to test the models
+# - actually run the computations
+
+# %%
+# Model definition
+# ================
+#
+# The machine learning model used here is a logistic regression.
+# The `warm_start` argument is essential is this example as it indicate to use the current state of the model
+# as initialization for the future training.
+# By default scikit-learn uses `max_iter=100`, which means the model trains on up to 100 epochs.
+# When doing FL, we don't want to train too much locally at every round
+# otherwise the local training will erase what were learned from the other centers. That is why we set `max_iter` to 3 here.
+
+
+import os
+from sklearn import linear_model
+
+cls = linear_model.LogisticRegression(random_state=SEED, warm_start=True, max_iter=3)
+
+# Optional:
+# Scikit-Learn raises warnings in case of non convergence, that we choose to disable here.
+# As this example runs with python subprocess, the way to disable it is to use environment
+# variable as below.
+
```

Comment:
```suggestion
# Scikit-Learn raises warnings in case of non convergence, that we choose to disable here.
# As this example runs with python subprocess, the way to disable it is to use following environment
# variable:
```